encoding/ascii.go

```diff
@@ -20,6 +22,9 @@ func (e asciiEncoder) Encode(data []byte) ([]byte, error) {
 
 func (e asciiEncoder) Decode(data []byte, length int) ([]byte, int, error) {
 	// read only 'length' bytes (1 byte - 1 ASCII character)
+	if len(data) < length {
+		return nil, 0, fmt.Errorf("not enought data to decode. expected len %d, got %d", length, len(data))
```

Comment:
bit of a spelling nit but `enought` -> `enough` :)

(and below)

Reply:
thanks for noticing it!